$OpenBSD: patch-mozilla_storage_src_mozStorageConnection_cpp,v 1.10 2012/09/01 14:10:03 landry Exp $
force the use of secure_delete instead of using internal sqlite copy
--- mozilla/storage/src/mozStorageConnection.cpp.orig	Mon Aug 27 06:49:30 2012
+++ mozilla/storage/src/mozStorageConnection.cpp	Sat Sep  1 02:00:15 2012
@@ -663,6 +663,13 @@ Connection::initialize(nsIFile *aDatabaseFile,
     return convertResultCode(srv);
   }
 
+  srv = ::sqlite3_exec(mDBConn, "PRAGMA secure_delete = ON", NULL, NULL, NULL);
+  if (srv != SQLITE_OK) {
+    ::sqlite3_close(mDBConn);
+    mDBConn = nsnull;
+    return convertResultCode(srv);
+  }
+
   // Set the synchronous PRAGMA, according to the preference.
   switch (Service::getSynchronousPref()) {
     case 2:
